Fix: variable redeclaration in `subghz_cli.c`

derskythe opened the pull request:
# What's new

- Variable `i` was redeclared within an inner loop, potentially leading to undesirable behavior. To prevent any issues, this redeclaration was corrected by changing the inner loop's variable name to `j`.

# Verification 

- No undesirable behavior
 
# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

derskythe committed:
Fix: variable redeclaration in `subghz_cli.c`

Variable `i` was redeclared within an inner loop, potentially leading to undesirable behavior. To prevent any issues, this redeclaration was corrected by changing the inner loop's variable name to `j`.

skotopes commented on applications/main/subghz/subghz_cli.c:
```diff
@@ -1067,7 +1067,7 @@ static void subghz_cli_command_chat(Cli* cli, FuriString* args) {
                         if(message[i] == '\n') {
                             printf("\r");
-                            for(uint8_t i = 0; i < 80; i++) {
+                            for(uint8_t j = 0; j < 80; i++) {
```
To be honest this specific place was just shadowing previous declaration without causing any errors, but now `i++` increments completely different thing.

derskythe replied:
@skotopes on the other hand, it still looks like a bug. 
But, if no one has any complaints, then it’s already a feature. 
But still I would add a comment with a link to the current pull request

skotopes replied:
Just fix `i++` thing )

derskythe replied:
@skotopes updated. As I understand this loop just create extra space to put line on the side.

derskythe committed:
fix: Improve newline handling in subghz_cli.c

Adjustments were made to ensure that newlines are correctly handled. The code now properly checks if the character before a newline is a carriage return, and if it's not, it adds it. Additional formatting is implemented to improve the visual layout of the output messages.

hedger committed:
Merge branch 'dev' into fix/shadow-variable

skotopes committed:
Merge branch 'dev' into fix/shadow-variable

skotopes commented on applications/main/subghz/subghz_cli.c:
```diff
@@ -1064,16 +1064,26 @@ static void subghz_cli_command_chat(Cli* cli, FuriString* args) {
-                            printf("\r");
-                            for(uint8_t i = 0; i < 80; i++) {
+                            // Check it was \r before
+                            if((i > 0 && message[i - 1] != '\r') || i == 0) {
```
this time you overdid yourself: now layout is a total mess. Check with `./fbt cli`

skotopes replied:
I mean what exactly you were trying to fix there?
